File: src/1921.cpp
#include <vector>
#include <algorithm>

using namespace std;

class Solution {
public:
    int eliminateMaximum(vector<int>& dist, vector<int>& speed) {
        for(int i = 0; i < dist.size(); ++i) {
            dist[i] = (dist[i] - 1) / speed[i];
        }
        sort(dist.begin(), dist.end());
        for(int i = 0; i < dist.size(); ++i) {
            if(dist[i] < i) {
                return i;
            }
        }
        return dist.size();
    }
};

File: src/1921.py
class Solution:
    def eliminateMaximum(self, dist: list[int], speed: list[int]) -> int:
        for i in range(len(dist)):
            if dist[i] % speed[i] == 0:
                dist[i] = (dist[i] // speed[i]) - 1
            else:
                dist[i] = dist[i] // speed[i]
        dist.sort()
        for i in range(len(dist)):
            if dist[i] < i:
                return i
        return len(dist)
